vcd/resource_vcd_catalog_vapp_template.go

```diff
@@ -159,32 +162,29 @@ func genericVcdCatalogVappTemplateRead(_ context.Context, d *schema.ResourceData
 	dSet(d, "name", vAppTemplate.VAppTemplate.Name)
 	dSet(d, "created", vAppTemplate.VAppTemplate.DateCreated)
 	dSet(d, "description", vAppTemplate.VAppTemplate.Description)
-	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
-	if err != nil {
-		return diag.Errorf(errorRetrievingOrg, err)
-	}
 
 	_, isCatalogIdSet := d.GetOk("catalog_id")
 	if !isCatalogIdSet { // This can only happen in the data source.
 		catalogName, err := vAppTemplate.GetCatalogName()
 		if err != nil {
 			return diag.Errorf("error retrieving the Catalog name to which the vApp Template '%s' belongs: %s", vAppTemplate.VAppTemplate.Name, err)
 		}
-		catalog, err := adminOrg.GetCatalogByName(catalogName, false)
+
+		orgName, err := vcdClient.GetOrgNameFromResource(d)
+		if err != nil {
+			return diag.Errorf("no org name found in resource data: %s", err)
+		}
+		catalog, err := vcdClient.Client.GetCatalogByName(orgName, catalogName)
 		if err != nil {
 			return diag.Errorf("error retrieving Catalog from vApp Template with name %s: %s", vAppTemplate.VAppTemplate.Name, err)
 		}
 		dSet(d, "catalog_id", catalog.Catalog.ID)
 	} else {
-		vdcName, err := vAppTemplate.GetVdcName()
+		vappTemplateRec, err := vAppTemplate.GetVappTemplateRecord()
 		if err != nil {
-			return diag.Errorf("error retrieving the VDC name to which the vApp Template '%s' belongs: %s", vAppTemplate.VAppTemplate.Name, err)
+			return diag.Errorf("error retrieving the vApp Template record for '%s': %s", vAppTemplate.VAppTemplate.Name, err)
 		}
-		vdc, err := adminOrg.GetVDCByName(vdcName, false)
-		if err != nil {
-			return diag.Errorf("error retrieving the VDC to which the vApp Template '%s' belongs: %s", vAppTemplate.VAppTemplate.Name, err)
-		}
-		dSet(d, "vdc_id", vdc.Vdc.ID)
+		dSet(d, "vdc_id", "urn:vcloud:vdc:"+extractUuid(vappTemplateRec.Vdc))
```

Comment:
Is this because `vappTemplateRec.Vdc` is the HREF?

Reply:
Yes. Usually, in query result, the name of the entity has the HREF ("owner" = owner HREF, "org" = org HREF, "vdc" = vdc HREF, and so on) while the field containing the name has a "Name" suffix.